File: 백준/Silver/1735. 분수 합/분수 합.py
def GCD(a, b):
    if b == 0:
        return a
    return GCD(b, a%b)

A_1, B_1 = map(int, input().split())
A_2, B_2 = map(int, input().split())

lcm = 0
if B_1 > B_2:
    lcm = B_1 * B_2 // GCD(B_1, B_2)
else:
    lcm = B_1 * B_2 // GCD(B_2, B_1)

A = lcm // B_1 * A_1 + lcm // B_2 * A_2


gcd = 0
if A > lcm:
    gcd = GCD(A, lcm)
else:
    gcd = GCD(lcm, A)
    
print(A//gcd, lcm//gcd)

File: 백준/Silver/1735. 분수 합/분수 합.java
import java.util.Scanner;

public class Main {

    public static int GCD(int a, int b) {
        if (b == 0) return a;
        return GCD(b, a % b);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int A1 = sc.nextInt();
        int B1 = sc.nextInt();
        int A2 = sc.nextInt();
        int B2 = sc.nextInt();

        int lcm;
        if (B1 > B2) {
            lcm = B1 * B2 / GCD(B1, B2);
        } else {
            lcm = B1 * B2 / GCD(B2, B1);
        }

        int A = (lcm / B1) * A1 + (lcm / B2) * A2;

        int gcd;
        if (A > lcm) {
            gcd = GCD(A, lcm);
        } else {
            gcd = GCD(lcm, A);
        }

        System.out.println((A / gcd) + " " + (lcm / gcd));

        sc.close();
    }
}
